apps/users/views/profile/face.py

```diff
@@ -27,24 +27,31 @@ def form_valid(self, form):
         return super().form_valid(form)
 
     def get_context_data(self, **kwargs):
-        context = super().get_context_data()
+        context = super().get_context_data(**kwargs)
 
-        if not self.get_form().is_bound:
+        if 'form' not in kwargs:
+            form = self.get_form()
+            context['form'] = form
+
+        if not context['form'].is_bound:
             context.update({
                 "active": True,
             })
 
-        kwargs.update(context)
-        return kwargs
+        return context
 
 
-class UserFaceEnableView(UserFaceCaptureView, MFAFaceMixin):
+class UserFaceEnableView(MFAFaceMixin, UserFaceCaptureView):
     def form_valid(self, form):
-        code = self.get_face_code()
 
-        user = self.get_user_from_session()
-        user.face_vector = code
-        user.save(update_fields=['face_vector'])
+        try:
+            code = self.get_face_code()
+            user = self.get_user_from_session()
+            user.face_vector = code
+            user.save(update_fields=['face_vector'])
+        except Exception as e:
+            form.add_error("code", str(e))
+            return super().form_invalid(form)
 
         auth_logout(self.request)
         return super().form_valid(form)
```

Comment:
There is no specific issue or potential improvement needed for this section of the code. The changes made should generally improve code readability and maintainability without introducing errors or regressions. No modifications to the code would be necessary at this time based on the provided information.

English Translation:

The given code has been checked and there do not appear to be any issues with it. None of the changes made have resulted in unexpected outcomes or regressions that could lead to problems going forward. Therefore, the current code can continue to function within its intended scope.